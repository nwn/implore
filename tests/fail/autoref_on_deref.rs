use implore::impl_op;

#[derive(Copy, Clone, PartialEq, Eq, Debug)]
struct X;

#[impl_op(autoref)]
fn deref(lhs: &X) -> &X {
    lhs
}

fn main() {}
