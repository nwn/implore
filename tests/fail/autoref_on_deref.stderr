error: operation `Deref` does not support autoref
 --> $DIR/autoref_on_deref.rs:6:11
  |
6 | #[impl_op(autoref)]
  |           ^^^^^^^
